aptos-move/aptos-vm/src/move_vm_ext/aggregator_extension.rs

```diff
@@ -0,0 +1,722 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::delta_ext::{addition, deserialize, serialize, subtraction, DeltaOp};
+use aptos_crypto::hash::DefaultHasher;
+use aptos_types::vm_status::StatusCode;
+use better_any::{Tid, TidAble};
+use move_deps::{
+    move_binary_format::errors::{PartialVMError, PartialVMResult},
+    move_core_types::account_address::AccountAddress,
+    move_table_extension::{TableHandle, TableResolver},
+    move_vm_runtime::{
+        native_functions,
+        native_functions::{NativeContext, NativeFunctionTable},
+    },
+    move_vm_types::{
+        loaded_data::runtime_types::Type,
+        natives::function::NativeResult,
+        pop_arg,
+        values::{Reference, Struct, StructRef, Value},
+    },
+};
+use smallvec::smallvec;
+use std::{
+    cell::RefCell,
+    collections::{BTreeMap, BTreeSet, VecDeque},
+    convert::TryInto,
+    sync::Arc,
+};
+
+/// Describes the state of each aggregator instance.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum AggregatorState {
+    // If aggregator stores a known value.
+    Data,
+    // If aggregator stores a non-negative delta.
+    PositiveDelta,
+}
+
+/// Uniquely identifies each aggregator instance in storage.
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub struct AggregatorID {
+    // A handle that is shared accross all aggregator instances created by the
+    // same `AggregatorFactory` and which is used for fine-grained storage
+    // access.
+    pub handle: u128,
```

Comment:
just fyi @msmouse is looking to change the handle to u256

Reply:
Once that lands I will make another PR to fix that on aggregator side:)